pkg/pusher/pusher_test.go

```diff
@@ -89,6 +89,50 @@ func TestSendChunkToPushSync(t *testing.T) {
 	p.Close()
 }
 
+// TestSendChunkToPushSyncWithoutTag is similar to TestSendChunkToPushSync, excep that the tags are not
+// present to simulate bzz api withotu splitter condition
+func TestSendChunkToPushSyncWithoutTag(t *testing.T) {
+	chunk := createChunk()
+
+	// create a trigger  and a closestpeer
+	triggerPeer := swarm.MustParseHexAddress("6000000000000000000000000000000000000000000000000000000000000000")
+	closestPeer := swarm.MustParseHexAddress("f000000000000000000000000000000000000000000000000000000000000000")
+
+	pushSyncService := pushsyncmock.New(func(ctx context.Context, chunk swarm.Chunk) (*pushsync.Receipt, error) {
+		receipt := &pushsync.Receipt{
+			Address: swarm.NewAddress(chunk.Address().Bytes()),
+		}
+		return receipt, nil
+	})
+	mtag := tags.NewTags()
+	_, err := mtag.Create("name", 1, false)
+	if err != nil {
+		t.Fatal(err)
+	}
+	p, storer := createPusher(t, triggerPeer, pushSyncService, mtag, mock.WithClosestPeer(closestPeer))
+	defer storer.Close()
+
+	_, err = storer.Put(context.Background(), storage.ModePutUpload, chunk)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	// Check is the chunk is set as synced in the DB.
+	for i := 0; i < noOfRetries; i++ {
+		// Give some time for chunk to be pushed and receipt to be received
+		time.Sleep(10 * time.Millisecond)
+
+		err = checkIfModeSet(chunk.Address(), storage.ModeSetSyncPush, storer)
+		if err == nil {
+			break
+		}
+	}
+	if err != nil {
+		t.Fatal(err)
+	}
+	p.Close()
```

Comment:
please convert to `defer` statement

Reply:
This is done to make sure that we close the usher before the storer. Otherwise we Add on the Wait in the storer.Close(). remember, we talked in length about this.